sympy/solvers/inequalities.py

```diff
@@ -1004,3 +1003,241 @@ def reduce_inequalities(inequalities, symbols=[]):
+    >>> d[z]
+    (x > z, z > Max(-x/2 - 13/10, -2*x/5 - 2/5))
+    """
+    pivot = _find_pivot(inequalities)
+    res = {}
+    while pivot != None:
+        maxs, mins, extra = _split_min_max(inequalities, pivot)
+        res[pivot] = (mins > pivot ,pivot > maxs)
+        inequalities = _merge_mins_maxs(maxs, mins) + extra
+        pivot = _find_pivot(inequalities)
+    return inequalities, res
+
+
+def _fourier_motzkin_extension(inequalities):
+    """Extension of the Fourier-Motzkin algorithm to the case where inequalities
+    do not contain variables that have at least two coefficients with opposite sign.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.inequalities import _fourier_motzkin_extension
+    >>> from sympy.abc import x, y, z
+
+    >>> eq1 = 2*x - 3*y + z + 1
+    >>> eq2 = x - y + 2*z - 2
+    >>> eq3 = x - y + 3*z + 4
+    >>> eq4 = x + z
+
+    >>> inequalities = [eq1, eq2, eq3, eq4]
+    >>> d = _fourier_motzkin_extension(inequalities)
+    >>> assert set(d) == {x}
+    >>> d[x]
+    (oo > x, x > Max(-z, y - 3*z - 4, y - 2*z + 2, 3*y/2 - z/2 - 1/2))
+    """
+
+    pivot = _pick_var(inequalities)
+    res = {}
+    while inequalities:
+        maxs, mins, extra = _split_min_max(inequalities, pivot)
+        res[pivot] = (mins > pivot ,pivot > maxs)
+        inequalities = extra
+        pivot = _pick_var(inequalities)
+    return res
+
+
+def _pick_var(inequalities):
+    """Return a free variable of the system of inequalities
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.inequalities import _pick_var
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.core.sorting import ordered
+
+    >>> eq1 = 2*x - 3*y + z + 1
+    >>> eq2 = x - y + 2*z - 2
+    >>> eq3 = x + y + 3*z + 4
+    >>> eq4 = x - z
+
+    >>> inequalities = [eq1, eq2, eq3, eq4]
+    >>> _pick_var(inequalities)
+    x
+    """
+    for eq in inequalities:  # should already be in canonical order
+        for symb in ordered(eq.free_symbols):  # make selection canonical
+            return symb
+
+def solve_linear_inequalities(inequalities):
+    """ Solve a system of linear inequalities
+
+    Parameters
+    ==========
+
+    inequalities: list of sympy equations
+        The system of inequalities to solve. All equations in the list are assume to be
+        linear and greater than 0. The system must be expressed as follows:
+
+        2x - 3y +  z + 1 > 0
+        x  -  y + 2z - 2 > 0
+        x  +  y + 3z + 4 > 0
+        x       -  z     > 0
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.inequalities import solve_linear_inequalities
+    >>> from sympy.abc import x, y, z
+    >>> from sympy.core.sorting import ordered
+
+    >>> eq1 = 2*x - 3*y + z + 1
+    >>> eq2 = x - y + 2*z - 2
+    >>> eq3 = x + y + 3*z + 4
+    >>> eq4 = x - z
+
+    >>> inequalities=[eq1, eq2, eq3, eq4]
+    >>> d = solve_linear_inequalities(inequalities)
+    >>> assert set(d) == set([x, y, z])
+    >>> d[x]
+    (oo > x, x > -2/7)
```

Comment:
Perhaps these should be `And` expressions like `And(oo>x,x<-2/7)`

Reply:
Ok, I will check that.

Reply:
I tried to replace the tuple by an And. The problem is that And seem to have it own order. With the tuple, we know that the first member is greater than the variable and that the second is lower. With an And from sympy, there is a risk that these positions switch.

Reply:
In the final result that doesn't matter since both conditions must be true and their sense is controlled by the type of the inequality. Am I missing something?

Reply:
It's easier to convert from lower and upper bounds to a Boolean than the other way around so I would rather have a function that returns bounds. There could also be a helper function that calls that and returns Booleans instead.